sql/driver/driver_test.go

```diff
@@ -45,18 +55,79 @@ func cleanup(s *server.TestServer, db *sql.DB) {
 
 func TestPlaceholders(t *testing.T) {
 	defer leaktest.AfterTest(t)
+
+	// Stolen from https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+	locationNames := []string{
+		"Canada/Newfoundland", // half-hour zone
+		"Europe/London",       // same as UTC, except for DST
+		"Pacific/Kiritimati",  // maximum positive offset
+		"Pacific/Midway",      // maximum negative offset
+		"US/Pacific",          // negative offset
+		"UTC",
+	}
+
 	s, db := setup(t)
 	defer cleanup(s, db)
 
-	year, month, day := 3015, time.August, 30
-	timeVal := time.Date(year, month, day, 3, 34, 45, 345670000, time.UTC)
-	dateVal := time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+	// We don't want to spend forever on this test.
```

Comment:
>  these tests would be easier to write as logic tests.

It's important to remember that the semantics of a logic test are different because the entire query is sent as a string and processed on the server, while placeholders are sent as protos encoded on the client. I'll revert this, though, and put the interesting combinations in their own stateless test.